components/brave_wallet_ui/common/async/interactionNotifier.ts

```diff
@@ -17,23 +18,27 @@ export default class InteractionNotifier {
     }
   }
 
-  beginWatchingForInteraction (timeMs: number, onInteractionInterval: () => unknown) {
-    this.intervalId = setInterval(
-      () => {
-        if (this.userHasInteracted) {
-          this.userHasInteracted = false
-          onInteractionInterval()
-        }
-      }, timeMs)
-    window.addEventListener('mousemove', this.handleInteraction)
-    window.addEventListener('keydown', this.handleInteraction)
-    window.addEventListener('focus', this.handleInteraction, true)
-    window.addEventListener('scroll', this.handleInteraction, true)
-    document.addEventListener('visibilitychange', this.handleVisibilityChange)
+  beginWatchingForInteraction (timeMs: number, isWalletLocked: boolean, onInteractionInterval: () => unknown) {
+    if (!isWalletLocked && !this.intervalAllreadyStarted) {
```

Comment:
I think a bit cleaner to not have `intervalAllreadyStarted` at all and just check if this.intervalId is set or not.  And clear that after clearing the timer.

Reply:
Updated! 👍